### PR TITLE
No shape noise

### PR DESCRIPTION
**Pull request by jpratmarti**
Fixing config file to run through with cosmodc2 and adding feature to run without shape noise.

### CODE REVIEW

File: examples/config/cosmodc2_config.yml [hunk 1]
```diff
@@ -85,13 +85,27 @@ TXSourceSelector:
...
 
 TXRandomCat:
     chunk_rows: 100000
     density: 10 # gals per sq arcmin
 
+TXJackknifeCenters:
+    npatch: 40
+
+TXDiagnosticPlots:
+    psf_prefix: mcal_psf_
```
Comment (joezuntz) [inline examples/config/cosmodc2_config.yml:99]: We weren't using the mcal PSFs because they are rounded - i.e. all the ellipticities are close to zero, whatever the actual PSF.

File: examples/config/cosmodc2_config.yml [hunk 2]
```diff
@@ -101,8 +115,21 @@ TXTwoPoint:
     max_sep: 250
     nbins: 20
     verbose: 0
+    shape_noise: True
 
-
+TXTwoPointLensCat:
```
Comment (joezuntz) [inline examples/config/cosmodc2_config.yml:120]: Is this related to this PR?

File: examples/config/cosmodc2_config.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ TXCosmoDC2Mock:
     flip_g2: True # to match metacal
 
 TXIngestRedmagic:
-    lens_zbin_edges: [0.0,0.2,0.4]
+    lens_zbin_edges: [0.1, 0.3, 0.5, 0.7, 0.9]
```
Comment (joezuntz) [inline examples/config/cosmodc2_config.yml:19]: Is this related to this PR?

File: txpipe/lens_selector.py [hunk 4]
```diff
@@ -260,24 +260,16 @@ class TXTruthLensSelector(TXBaseLensSelector):
     def data_iterator(self):
         print(f"We are cheating and using the true redshift.")
         chunk_rows = self.config['chunk_rows']
-        if self.config['true_z']:
```
Comment (joezuntz) [inline txpipe/lens_selector.py:263]: Thanks for simplifying this!

File: txpipe/plotting/correlations.py [hunk 5]
```diff
@@ -133,13 +133,13 @@ def axis_setup(a, i, j, ny, ymin, ymax, name):
...
     else:
-        a.set_xlabel(r"$\theta / $ arcmin")
+        a.set_xlabel(r"$\theta$ [arcmin]")
 
     a.tick_params(axis='both', which='major', length=10, direction='in')
     a.tick_params(axis='both', which='minor', length=5, direction='in')
 
     # Fix
-    a.text(0.1, 0.1, f"Bin {i}-{j}", transform=a.transAxes)
+    a.text(0.1, 0.1, f"{j+1} - {i+1}", transform=a.transAxes)
```
Comment (joezuntz) [inline txpipe/plotting/correlations.py:142]: We had a vote on the 3x2pt channel and in LSST bins officially start from zero!
https://lsstc.slack.com/archives/C9JSJV3FZ/p1521045279000638

File: txpipe/plotting/correlations.py [hunk 6]
```diff
@@ -352,7 +353,13 @@ def make_theory_plot_data(data, cosmo, obs, label, smooth=True, xi=None):
             # Optionally also compute correlation functions
             if xi:
                 theta, *_ = obs[(GAMMA, i, j)]
-                theory[GAMMA, i, j] = theta, pyccl.correlation(cosmo, ell, cl, theta/60, corr_type='GL')
```
Comment (joezuntz) [inline txpipe/plotting/correlations.py:355]: I fixed this in CCL.

File: txpipe/twopoint.py [hunk 7]
```diff
@@ -52,6 +52,7 @@ class TXTwoPoint(PipelineStage):
         'do_shear_pos': True,
         'do_pos_pos': True,
         'var_methods': 'jackknife',
+        'shape_noise': True,
```
Comment (joezuntz) [inline txpipe/twopoint.py:55]: could you change the name of this to something that makes clear it's purpose, like `use_true_shear`?

File: txpipe/plotting/correlations.py (hunk 5 above)
Comment (jpratmarti) [reply txpipe/plotting/correlations.py:142]: Oh sorry about this I had no idea! Will correct it and push it again.

File: txpipe/plotting/correlations.py (hunk 6 above)
Comment (jpratmarti) [reply txpipe/plotting/correlations.py:355]: Great! so then this means I should remove this if statement I added right?

File: examples/config/cosmodc2_config.yml (hunk 1 above)
Comment (jpratmarti) [reply examples/config/cosmodc2_config.yml:99]: I only added this because it was raising an error otherwise.. I'll remove it for the moment though.

File: examples/config/cosmodc2_config.yml (hunk 2 above)
Comment (jpratmarti) [reply examples/config/cosmodc2_config.yml:120]: When using `cosmodc2_redmagic_pipeline.yml`, it requires to have that stage defined I think so I added this there to fix it.

File: examples/config/cosmodc2_config.yml (hunk 3 above)
Comment (jpratmarti) [reply examples/config/cosmodc2_config.yml:19]: Not specifically, but the [0.0,0.2,0.4] edges didn't make much sense given the N(z) of the sample so I changed it. I can change it back though, as you prefer :)

File: txpipe/twopoint.py (hunk 7 above)
Comment (jpratmarti) [reply txpipe/twopoint.py:55]: Done.

File: examples/config/cosmodc2_config.yml (hunk 3 above)
Comment (joezuntz) [reply examples/config/cosmodc2_config.yml:19]: no, this is useful
